backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/utils/Geometry.java

```diff
@@ -12,16 +12,48 @@
  *******************************************************************************/
 package org.eclipse.sirius.components.diagrams.layout.incremental.utils;
 
+import java.util.EnumMap;
+import java.util.Map.Entry;
+import java.util.Objects;
 import java.util.Optional;
 
 import org.eclipse.sirius.components.diagrams.Position;
+import org.eclipse.sirius.components.diagrams.Size;
 
 /**
  * A Geometry Helper.
  *
  * @author fbarbin
  */
 public final class Geometry {
+    /**
+     * Represents the 4 different sides of a rectangle.
+     */
+    public enum RectangleSide {
+        NORTH, EAST, SOUTH, WEST
+    }
```

Comment:
Extract this to a proper file. This is not an internal concept since it's manipulated outside of this class.

Reply:
done